static/src/javascripts-legacy/projects/common/modules/experiments/tests/acquisitions-thrasher-uk-election.js

```diff
@@ -0,0 +1,99 @@
+define([
+    'lib/$',
+    'lib/element-inview',
+    'lib/storage',
+    'commercial/modules/user-features',
+    'common/modules/commercial/contributions-utilities',
+    'ophan/ng'
+], function (
+    $,
+    elementInView,
+    storage,
+    userFeatures,
+    contributionUtilities,
+    ophan
+) {
+
+    var UK_ELECTION_THRASHER_VIEW_COUNTER = "gu.uk-election-thrasher.views";
+    var UK_ELECTION_THRASHER_BLOCK_ELEMENT = "election-supporters__wrapper";
+
+    function getThrasherViewedCount() {
+        return parseInt(storage.local.get(UK_ELECTION_THRASHER_VIEW_COUNTER)) || 0;
+    }
+
+    function incrementThrasherViewedCount() {
+        storage.local.set(UK_ELECTION_THRASHER_VIEW_COUNTER, getThrasherViewedCount() + 1)
+    }
+
+    var $ukElectionThrasher =  $("." + UK_ELECTION_THRASHER_BLOCK_ELEMENT);
+
+    function isThankYouVariantReader() {
+        return userFeatures.isPayingMember() || contributionUtilities.isContributor
+    }
+
+    function recordThrasherVariant() {
+        // Enables the conversion rate of the ask variant to be calculated.
+        ophan.record({
+            component: 'uk_election_thrasher_variant',
+            value: isThankYouVariantReader() ? "thank_you" : "ask"
+        })
+    }
+
+    function getReaderSpecificUkElectionThrasherClassList() {
+        // There are eight variants of the thank you thrasher prefixed by -0, -1, -2, ..., -7
+        var viewCountClass = UK_ELECTION_THRASHER_BLOCK_ELEMENT + "--" + getThrasherViewedCount() % 8;
+        var thrasherVariantClass =  UK_ELECTION_THRASHER_BLOCK_ELEMENT + "--" + (isThankYouVariantReader() ? "thanks" : "ask");
+        return viewCountClass + " " + thrasherVariantClass
+    }
+
+    function onThrasherViewed(callback) {
+        // Safe selecting from array - function will only be called if length of array element is positive.
+        // Element in view logic taken from contribution utilities.
+        elementInView($ukElectionThrasher[0], window, { top: 18 }).on('firstview', callback())
+    }
+
+    return function() {
+
+            this.id = 'AcquisitionsThrasherUkElection';
+
```

Comment:
Correct - this field is introduced by contribution utilities which I'm not using.